### DISCUSSION TITLE
add c3c in github-linguist

### DISCUSSION THREAD

**Discussion by ghost** [c3lang/c3c#573] [Q&A] [answered]
Hi! 

I was thinking of adding an issue on [GitHub](https://github.com/github/linguist) to identify `.c3c` files that are in the c3lang language. 

What do you all think of this idea?

#### 1. Example issue: *add c3c in github-linguist*
```markdown

<!--

👋 If you have the time and know-how, send us a pull-request: everybody is welcome to contribute! 
Otherwise, fill out the following fields as best you can.

-->

## Language name
c3lang

## URL of example repository
https://github.com/c3lang/c3c

## URL of syntax highlighting grammar
http://www.c3-lang.org/

## Most popular extensions
`.c3`

<!--

List the extensions commonly used by this language.

-->

## Detected language
(None)

<!--

What language are files for this language being identified as, if any?

-->

```

#### 2. main.c3 [c3-lang/examples](http://www.c3-lang.org/examples/)

```c
fn void example_foreach(float[] values) 
{
    foreach (index, value : values) 
    {
        libc::printf("%d: %f\n", index, value);
    }
}
```

**Comment by data-man** [comment, marked as answer]
According to https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language

> In most cases we prefer that each new file extension be in use in **at least 200 unique :user/:repo repositories** before supporting them in Linguist

But you can try! :)

**Reply by ghost** [reply]
Hi data-man. thank you for feedback, I'll try anyway.

**Comment by lerno** [comment]
And the suffix is .c3 😅

**Reply by ghost** [reply]
Hi lerno. thank you for feedback - I will make this fix.

**Comment by lerno** [comment]
And you need a js highlighter I think

**Reply by ghost** [reply]
Hi. Like this: [highlightjs](https://highlightjs.org/) ? *" Syntax highlighting for the Web" - highlightjs*

**Reply by lerno** [reply]
Hmm... looking at it.. no, you need a textmate compatible grammar.

**Comment by lerno** [comment]
And the entry will end up looking something like
```
C3:
  type: programming
  color: "#555555"
  extensions:
  - ".c3"
  - ".c3t"
  tm_scope: source.c3
```

**Reply by lerno** [reply]
I mean the entry in languages.yml, not the grammar.
